ocp/cosa-image.yaml

```diff
@@ -0,0 +1,220 @@
+# In this file, you will find the following:
+#   - ImageStream configs for importing Fedora images
+#   - ImageStream configs for importing coreos-assembler images
+#   - BuildConfig defining how coreos-assembler:master is built
+#   - BuildConfig defining how coreos-assembler:buildconfig is built
+#
+# To use this file, you would do something like the following:
+#   $ oc apply -f cosa-image.yaml
+#
+# It is possible to provide parameters used by the BuildConfig, e.g.:
+#   $ oc process -f cosa-image.yaml --param=REPO_REF=foo \
+#                                   --param=NAMESPACE=bar | \
+#     oc apply -f -
+#
+apiVersion: v1
+kind: Template
+labels:
+  app: cosa
+  template: cosa-template
+metadata:
+  annotations:
+    description: CoreOS Assembler Images
+    openshift.io/display-name: CoreOS Assembler
+    tags: cosa,coreos-assembler
+  name: cosa-images
+
+parameters:
+  - description: Git source URI
+    name: REPO_URL
+    value: "https://github.com/darkmuggle/coreos-assembler"
+  - description: Git branch/tag reference
+    name: REPO_REF
+    value: "psi"
+  - description: Namespace of the pipeline
+    value: ${NAMESPACE}
+    name: NAMESPACE
+  - description: Name of the image; use e.g. joe-cosa for your own build
+    name: IMAGE
+    value: coreos-assembler
+
+objects:
+
+  ### Fedora Caching ###
+  # keep a local copy of Fedora, since its use to build our Docker Images
+  - apiVersion: "image.openshift.io/v1"
+    kind: ImageStream
+    metadata:
+      name: fedora
+      labels:
+        app: cosa
+        template: cosa-template
+    spec:
+      failedBuildHistoryLimit: 1
+      successfulBuildsHistoryLimit: 1
+      lookupPolicy:
+        # this allows e.g. the pipeline to directly reference the imagestream
+        local: true
+      tags:
+      - from:
+          kind: DockerImage
+          name: 'registry.fedoraproject.org/fedora:32'
+        name: '32'
+        referencePolicy:
+          type: Source
+        importPolicy:
+          scheduled: true
+
+  ### COREOS-ASSEMBLER Local Image Stream ###
+  # This is the image stream that local builds push to
+  - apiVersion: image.openshift.io/v1
+    kind: ImageStream
+    metadata:
+      name: coreos-assembler
+    spec:
+      lookupPolicy:
+        local: true
+      failedBuildHistoryLimit: 1
+      successfulBuildsHistoryLimit: 1
+      lookupPolicy:
+        local: true
+
+  ### COREOS-ASSEMBLER Import Stream
```

Comment:
This one I'm also uncertain to have in cosa itself, we're just using this as a hacky way to retrigger rhcos cosa builds when quay changes - it's not clear to me everyone will want that, and it feels circular to have in upstream git right?

Reply:
Yeah.. I'd call this a devel artifact. Circular, yes. But Quay is not multi-arch, so this is useful at least in that world. Maybe? I *think* I'll drop this from the next push.